internal/controllers/clusterextension_controller.go

```diff
@@ -112,82 +179,232 @@ func checkForUnexpectedFieldChange(a, b ocv1alpha1.ClusterExtension) bool {
 	return !equality.Semantic.DeepEqual(a, b)
 }
 
+func (r *ClusterExtensionReconciler) handleResolutionErrors(ext *ocv1alpha1.ClusterExtension, err error) (ctrl.Result, error) {
+	var aggErrs utilerrors.Aggregate
+	if errors.As(err, &aggErrs) {
+		for _, err := range aggErrs.Errors() {
+			errorMessage := err.Error()
+			if strings.Contains(errorMessage, "no package") {
+				// Handle no package found errors, potentially setting status conditions
+				setResolvedStatusConditionFailed(&ext.Status.Conditions, errorMessage, ext.Generation)
+				ensureAllConditionsWithReason(ext, "ResolutionFailed", errorMessage)
+			} else if strings.Contains(errorMessage, "invalid version range") {
+				// Handle invalid version range errors, potentially setting status conditions
+				setResolvedStatusConditionFailed(&ext.Status.Conditions, errorMessage, ext.Generation)
+				ensureAllConditionsWithReason(ext, "ResolutionFailed", errorMessage)
+			} else {
+				// General error handling
+				setResolvedStatusConditionFailed(&ext.Status.Conditions, errorMessage, ext.Generation)
+				ensureAllConditionsWithReason(ext, "InstallationStatusUnknown", "")
```

Comment:
@bentito Do we still need this func? Can we set the resolution error directly and test the returned string - as we did before?

Reply:
If we do need it - would be great to replace `"ResolutionFailed"` and `"InstallationStatusUnknown"` with constants from `api/v1alpha1/clusterextension_types.go`.

Reply:
Also - why unknown here and below? We know that resolution failed.

In general - it looks like we can get rid of this and `ensureAllConditionsWithReason`. I imagine that we can just do something like that:

```go
	bundle, err := r.resolve(ctx, *ext)
	if err != nil {
		setResolvedStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.Generation)
		// setInstalledStatusConditionFailed, etc
		return ctrl.Result{}, err
	}
```

Reply:
Error handling will be resolved in PR #878.